xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -81,12 +85,13 @@ public String getServiceAuthority() {
   @SuppressWarnings("unchecked")
   @Override
   public void start(final Listener2 listener) {
-    BootstrapInfo bootstrapInfo = null;
-    try {
-      bootstrapInfo = bootstrapper.readBootstrap();
-    } catch (Exception e) {
-      listener.onError(Status.UNAVAILABLE.withDescription("Failed to bootstrap").withCause(e));
-      return;
+    if (bootstrapInfo == null) {
```

Comment:
Is there any reason to put this check? Even if `bootstrapInfo != null` you can still `readBootstrap` thereby getting the latest content (unless there is a requirement that you should read the bootstrap just once for each `XdsNameResolver` - I thought that was more of an accommodation rather than a requirement).

Reply:
Sure, I agree. I've been completely overthink this problem. Neither C++ nor Go has any caching mechanism for reading bootstrap file.